test/result_error_move.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright 2023 Klemens Morgenstern
+// Distributed under the Boost Software License, Version 1.0.
+// https://www.boost.org/LICENSE_1_0.txt
+
+#include <boost/system/result.hpp>
+#include <boost/core/lightweight_test.hpp>
+#include <boost/core/lightweight_test_trait.hpp>
+#include <string>
+
+using namespace boost::system;
+
+struct XM
+{
+    int v_;
+
+    explicit XM( int v = 0 ): v_( v ) {}
+
+    XM( XM const& ) = default;
```

Comment:
```suggestion
    XM( XM const& ) = delete;
```

This should've been `delete`.